lib/std/testing.zig

```diff
@@ -719,17 +762,20 @@ pub fn checkAllAllocationFailures(backing_allocator: std.mem.Allocator, comptime
     }
 }
 
-/// Given a type, reference all the declarations inside, so that the semantic analyzer sees them.
+/// Given a type, references all the declarations inside, so that the semantic analyzer sees them.
 pub fn refAllDecls(comptime T: type) void {
-    if (!builtin.is_test) return;
+    checkScope();
```

Comment:
This change in particular seems excessive; I'm pretty sure `refAllDecls` is meant to be easy enough to just stick in a block like
```
comptime {
    std.testing.refAllDecls(@This());
}
```
and leave it as is, without having to go through the ceremony of also wrapping it in a check for `std.builtin.is_test`.
Ditto for `refAllDeclsRecursive`.

Reply:
Yes, that's a good point. I was thinking about that too. I'm sure people do this. I went back a bit in history and apparently `refAllDecls` and or `refAllDeclsRecursive` actually used to be in `std.meta`. I'm thinking:
* Just use `test` instead of `comptime`
* Let's move `refAllDecls` and `refAllDeclsRecursive` back into `std.meta`.
  But then again, these are definitely for testing and shouldn't be required apart from that so maybe, yes, you should just use `test` instead of `comptime`?

Reply:
So now people have to do
```zig
test {
    std.testing.refAllDecls(X);
}
```
and I think that's better actually because it makes the intent clearer. You aren't just trying to do something at `comptime`, you're trying to `test` and make sure something works properly.

However I noticed you used `@This()` in your example. I think in this case people will just have to use it a bit differently, too.